main.go

```diff
@@ -93,6 +93,21 @@ func main() {
 	recorder := record.NewBroadcaster().NewRecorder(clientgoscheme.Scheme, v1.EventSource{Component: controllers.ComponentName})
 	kubeClient := kubernetes.NewForConfigOrDie(rest.AddUserAgent(config, controllers.ComponentName))
 
+	// Check the Platform Type to determine the state of the CO
+	enabled, err := controllers.IsEnabled(mgr.GetClient())
+	if err != nil {
+		setupLog.Error(err, "unable to determine Infrastructure Platform type")
+		os.Exit(1)
+	}
+	if !enabled {
+		// set ClusterOperator status to disabled=true, available=true
+		err = controllers.SetCOInDisabledState(osClient, releaseVersion)
+		if err != nil {
+			setupLog.Error(err, "unable to Baremetal CO to disabled")
```

Comment:
is a verb missing here?

Reply:
Also, is it ok to exit in such case?

Reply:
Fixed the verb.

Regarding whether to continue with setup of the operator or exit was not a clear choice for me.
If the CO is not updated correctly, the CVO considers the Operator to be in a bad state anyways. And in the case of non-Baremetal platforms, this code will run only once so there is no path where this will correct itself (which can happen when we update the CO in the reconcile methond).

Please check the "Althernate approach" that I listed above and let me know what you think.

Reply:
My feeling is that the alternate approach could be a little bit more clear. The resource will be amended within a reconcile loop, which could be a better place than the main entry point of the operator.